tests/nonce/test_nonce_utils.cairo

```diff
@@ -0,0 +1,89 @@
+use starknet::{
+    ContractAddress, get_caller_address, Felt252TryIntoContractAddress, contract_address_const
+};
+use snforge_std::{declare, start_prank, stop_prank, ContractClassTrait};
+
+use satoru::data::data_store::{IDataStoreSafeDispatcher, IDataStoreSafeDispatcherTrait};
+use satoru::role::role_store::{IRoleStoreSafeDispatcher, IRoleStoreSafeDispatcherTrait};
+use satoru::role::role;
+use satoru::nonce::nonce_utils::{get_current_nonce, increment_nonce, get_next_key};
+
+#[test]
+fn test_nonce_utils() {
+    // *********************************************************************************************
+    // *                              SETUP                                                        *
+    // *********************************************************************************************
+    let data_store = setup();
+
+    // *********************************************************************************************
+    // *                              TEST LOGIC                                                   *
+    // *********************************************************************************************
+
+    let nonce = get_current_nonce(data_store).unwrap();
```

Comment:
need to unwrap here because tests can't return a `Result` type, however normal calling code should use the following syntax:

```rust
let nonce = get_current_nonce(data_store)?;
```